.github/workflows/publish-types.yaml

```diff
@@ -0,0 +1,28 @@
+name: Publish Types
+
+on:
+  release:
+    types: [created, published]
+  push:
+    branches:
+      - staging
```

Comment:
We don't have a staging branch. The main branch acts as our staging.

Reply:
types: [created, published]
  push:
    branches:
      - main